src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -178,17 +185,20 @@ public void checkExistence() {
             // Handle response if valid itineraries exist.
             if (plan != null && plan.itineraries != null) {
                 for (Itinerary itineraryCandidate : plan.itineraries) {
-                    // If a matching itinerary is found, save the date with the matching itinerary.
-                    // The matching itinerary will replace the original trip.itinerary.
-                    if (ItineraryUtils.itinerariesMatch(referenceItinerary, itineraryCandidate)) {
-                        result.handleValidDate(otpRequest.date, itineraryCandidate);
-                        hasMatchingItinerary = true;
+                    // Make sure itinerary is same day as request date
+                    if (ItineraryUtils.occursOnSameServiceDay(itineraryCandidate, otpRequest.dateTime, tripIsArriveBy)) {
+                        // If a matching itinerary is found, save the date with the matching itinerary.
+                        // The matching itinerary will replace the original trip.itinerary.
+                        if (ItineraryUtils.itinerariesMatch(referenceItinerary, itineraryCandidate)) {
```

Comment:
Seems like these could be combined into a single if statement

Reply:
https://github.com/ibi-group/otp-middleware/pull/102/commits/98f91c7d7043e0039a39a1855dc3e40befbf178e.